lib/array_equals.rb

```diff
@@ -1,5 +1,26 @@
 # Determines if the two input arrays have the same count of elements
 # and the same integer values in the same exact order
 def array_equals(array1, array2)
-  raise NotImplementedError
+  i = 0
+
+  if (array1 == nil && array2 == nil) || (array1 == [] && array2 == [])
+    return true
+  elsif array1 == nil || array2 == nil
+    return false
+  end
+
+  if array1.length == array2.length
+    until i > array1.length
+      if array1[i] != array2[i]
+        return false
+      end
+      i += 1
+    end
+    return true
+  end
+  return false
 end
+
+array3 = [10, 20, 30, 40, 50, 60]
```

Comment:
Nice testing! :smile:

Aside: typically we'll put our tests in a separate file. This is
important because some languages (e.g. Python and Javascript) _automatically_
execute top-level code in loaded files/libraries.

e.g. if `fileB` imports `fileA`, top-level code (that's not in a function)
in `fileA` will automatically be executed in such languages. (In some cases, this **is** what you want - in others, it **isn't**.)